pkg/config/api_config.go

```diff
@@ -237,12 +249,93 @@ func LoadAPIConfigFromFile(path string) (*APIConfig, error) {
 	if err != nil {
 		return nil, perrors.Errorf("unmarshalYmlConfig error %v", perrors.WithStack(err))
 	}
-	once.Do(func() {
-		apiConfig = apiConf
-	})
+	apiConfig = apiConf
 	return apiConf, nil
 }
 
+// LoadAPIConfig load the api config from config center
+func LoadAPIConfig(metaConfig *model.APIMetaConfig) (*APIConfig, error) {
+
+	client = etcdv3.NewConfigClient(
+		etcdv3.WithName(etcdv3.RegistryETCDV3Client),
+		etcdv3.WithTimeout(10*time.Second),
+		etcdv3.WithEndpoints(strings.Split(metaConfig.Address, ",")...),
+	)
+
+	go listenAPIConfigNodeEvent(metaConfig.APIConfigPath)
+
+	content, err := client.Get(metaConfig.APIConfigPath)
+
+	if err != nil {
+		return nil, perrors.Errorf("Get remote config fail error %v", err)
+	}
+
+	initAPIConfigFromString(content)
+
+	return apiConfig, nil
+}
+
+func initAPIConfigFromString(content string) error{
+	lock.Lock()
+	defer lock.Unlock()
+
+	apiConf := &APIConfig{}
+	if len(content) != 0 {
+		err := yaml.UnmarshalYML([]byte(content), apiConf)
+		if err != nil {
+			return perrors.Errorf("unmarshalYmlConfig error %v", perrors.WithStack(err))
+		}
+		apiConfig = apiConf
+	}
+	return nil
+}
+
+func listenAPIConfigNodeEvent(key string) bool {
```

Comment:
can we have unit test for this function?

Reply:
I will add